src/OpenLineageWriter.php

```diff
@@ -33,15 +35,23 @@ public function write(): void
             }
             $this->logger->info(sprintf('Job %s import to OpenLineage API - start', $job['id']));
 
-            $lineAgeResponse = $this->getJobLineage($job['id']);
-            foreach ($lineAgeResponse as $event) {
+            $lineageResponse = $this->getJobLineage($job['id']);
+            foreach ($lineageResponse as $event) {
                 if ($this->jobNameAsConfig) {
                     $event['job']['name'] = sprintf('%s-%s', $job['component'], $job['config']);
                 }
                 $this->logger->info(sprintf('- Sending %s event', $event['eventType']));
-                $this->openLineageClient->request('POST', '/api/v1/lineage', [
-                    'body' => json_encode($event),
-                ]);
+
+                try {
+                    $this->openLineageClient->request('POST', '/api/v1/lineage', [
+                        'body' => json_encode($event),
+                    ]);
+                } catch (RequestException $e) {
+                    if (str_contains($e->getMessage(), 'cURL error 3:')) {
+                        throw new UserException('Malformed URL of OpenLineage server');
+                    }
+                    throw new UserException($e->getMessage());
+                }
```

Comment:
a nemělo by to odchytávat teda i nějaký 500 a řict, že se to nepovedlo?

Reply:
Nemyslim si. 500 ako odpoved zo serveru to tiez zachyti a prekonvertuje na UserException.

A kebyze je tam nejaka ina exception ako Guzzle RequestException tak ta prebubla dalej a skonci ako ApplicationException.

Reply:
tak ta prebubla dalej a skonci ako ApplicationException. -> no právě, čili když ten marquez pošle nějakou píčovinu, tak se user nic nedozví - já bych byl celkem pro to to tady probublat k userovi, protože je to jeho server

Reply:
Myslis to tak, ze pridat catch na Throwable a z toho spravit UserException? 
Proc ne, je to ako pises, je to uzivatelov server.

Reply:
jj

Reply:
OK, spravil som to tak. Ja si myslim, ze Guzzle vsetko preklopi na RequestException (alebo nejakeho potomka tejto triedy) ale preistotu som tam pridal aj ten Throwable a este jeden test (ktory ale testuje server error response, nie ten "Throwable") :)